syntax = "proto3";

import "service/DataService.proto";
package apb.api.consumableitem;

option java_package = "jp.co.apb.server.api.proto.consumableitem";
option java_multiple_files = true;
option csharp_namespace = "Art.Framework.ApiNetwork.Grpc.Api.ConsumableItem";

message UseEffectItemRequest {
  int32 consumable_item_id = 1;
  int32 count = 2;
}

message UseEffectItemResponse {
  map<string, .apb.api.data.DiffData> diff_user_data = 99;
}

message SellRequest {
  repeated .apb.api.consumableitem.SellPossession consumable_item_possession = 1;
}

message SellPossession {
  int32 consumable_item_id = 1;
  int32 count = 2;
}

message SellResponse {
  map<string, .apb.api.data.DiffData> diff_user_data = 99;
}

service ConsumableItemService {
  rpc UseEffectItem(.apb.api.consumableitem.UseEffectItemRequest) returns (.apb.api.consumableitem.UseEffectItemResponse);
  rpc Sell(.apb.api.consumableitem.SellRequest) returns (.apb.api.consumableitem.SellResponse);
}

